storage/filesystem/object.go

```diff
@@ -71,7 +71,14 @@ func (s *ObjectStorage) Get(h core.Hash) (core.Object, error) {
 	p := packfile.NewParser(r)
 
 	obj := s.NewObject()
-	return obj, p.FillObject(obj)
+	err = p.FillObject(obj)
+	if err != nil {
+		return nil, err
+	}
+	if core.AnyObject != t && obj.Type() != t {
```

Comment:
We don't know the type until we apply FillObject.
We could, however, call p.ReadObjectTypeAndLength() if we want to retrieve the type before full parsing. That would be more efficient for AnyObject and only slightly less efficient for the others.
What do you think?